frontend/main-app/src/components/FormViewer/FormViewer.tsx

```diff
@@ -0,0 +1,66 @@
+import { Alert, Button, Heading } from "@digdir/design-system-react";
+import React, { FormEvent, useState } from "react";
+import classes from "./FormViewer.module.css";
+import { TasklistSendFillIcon } from "@navikt/aksel-icons";
+import { FormRadio } from "../FormRadio";
+import { FormTextfield } from "../FormTextfield";
+import formSchema from "./formSchema";
+import { cleanFormData, generateValidationSchema } from "./FormViewerUtils";
+import { useTranslation } from "react-i18next";
+
+export const FormViewer = (): React.JSX.Element => {
+  const {t} = useTranslation();
+  const [formErrors, setFormErrors] = useState({});
+  const [formAlert, setFormAlert] = useState("");
+
+  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const formData = new FormData(event.currentTarget as HTMLFormElement);
+    const cleanedFormData = cleanFormData(formData);
+    const validationSchema = generateValidationSchema(formSchema);
+    const result = validationSchema.safeParse(cleanedFormData);
+
+    if ("error" in result) {
+      setFormErrors(result.error.formErrors.fieldErrors);
+      setFormAlert("error")
+    } else {
+      setFormErrors({});
+      // In the future: pass form submission to backend
+      setFormAlert("success");
+    }
+  };
+  return (
+    <main className={classes.card}>
+      <form onSubmit={handleSubmit}>
+        <Heading level={1} size="xlarge">
+          {formSchema.title}
+        </Heading>
+        {formSchema.components.map((c) =>
+          c.componentType === "textfield" ? (
```

Comment:
Just a suggestion, what about extracting this logic out, i.e. above the return (or in its own file), so that the props can be destructured, and make it easier to see what it contains.
I.e: 

```
{formSchema.components.map((component) => 
    {FormComponent("component")}
 }
 ```
 
 And then above the return:
 ```
 const FormComponent = (compProps) => {
     const {componentType, id as id.string(), question, required, etc } = compProps
     
     componentType ==="textfield" ? 
         <div key={id} className={classes.component}>
              <FormTextfield
                name={id}
                question={question}
                required={required}
                error={formErrors?.[id]}
              />
            </div>
            : <div ....

Reply:
I made a new React component `FormComponents` that handles this.